src/Http/Middleware/BodyParserMiddleware.php

```diff
@@ -182,7 +182,15 @@ public function process(ServerRequestInterface $request, RequestHandlerInterface
      */
     protected function decodeJson(string $body)
     {
-        return json_decode($body, true);
+        if ($body === '') {
+            return [];
+        }
+        $decoded = json_decode($body, true);
+        if (json_last_error() === JSON_ERROR_NONE) {
+            return (array)$decoded;
+        }
+
+        return false;
```

Comment:
Shouldn't this return null?

Reply:
Ideally yes, but I guess for BC reasons it should return similar values as pure json_decode.

Reply:
> NULL is returned if the json cannot be decoded

https://www.php.net/manual/en/function.json-decode.php

I thought it does.

Reply:
With the current code any non-array response will be considered a bad response. To keep more compatible with `json_encode` I updated it to `null`.

Reply:
Should we update the return type hint to `array|null` ?

Reply:
Done too.